### PR TITLE
Make Delete not return ErrNotFound for a missing key, per Datastore spec

### LINKED ISSUE #62: Delete should not return error for key not found
go-datastore's [Write](https://godoc.org/github.com/ipfs/go-datastore#Write) interface specifies for Delete:

```
    // Delete removes the value for given `key`. If the key is not in the
    // datastore, this method returns no error.
```

Yet we see in the local `Delete` method here: https://github.com/ipfs/go-ds-crdt/blob/master/crdt.go#L676

```
	if len(delta.Tombstones) == 0 {
		return ds.ErrNotFound
	}
```

This seems like a violation of the spec.

### PR DESCRIPTION
**Pull request by ncabatoff**
Fixes #62.  (Also a few typos.)

### CODE REVIEW

File: heads.go [hunk 1]
```diff
@@ -56,11 +56,7 @@ func (hh *heads) write(store ds.Write, c cid.Cid, height uint64) error {
 }
 
 func (hh *heads) delete(store ds.Write, c cid.Cid) error {
-	err := store.Delete(hh.key(c))
-	if err == ds.ErrNotFound {
-		return nil
-	}
-	return err
```
Comment (hsanjuan) [inline heads.go:63]: I'm not sure we should change this. This changes internal behaviour which was explicit in detecting that we are trying to remove an unexistent head. I don't know if this check is warranted or not (without diving deeper), but it is not related to this PR and I don't want to risk a subtle bug being introduced.

Comment (ncabatoff) [reply heads.go:63]: Sure, I can remove this part.

File: heads.go [hunk 2]
```diff
@@ -56,7 +56,14 @@ func (hh *heads) write(store ds.Write, c cid.Cid, height uint64) error {
 }
 
 func (hh *heads) delete(store ds.Write, c cid.Cid) error {
-	return store.Delete(hh.key(c))
+	err := store.Delete(hh.key(c))
+	// The go-datastore API currently says Delete doesn't return
+	// ErrNotFound, but it used to say otherwise.  Leave this
+	// here to be safe.
```
Comment (hsanjuan) [inline heads.go:62]: This is misleading. `heads.delete` is in no way trying to implement the go-datastore API.

Comment (hsanjuan) [reply heads.go:62]: The `heads` type is internal and is doing internal things, it's not a go-datastore instance.

Comment (hsanjuan) [reply heads.go:62]: Re-reading, I understand the comment now.

**Pull request merged**